catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/CachedClientPool.java

```diff
@@ -65,7 +65,11 @@ public class CachedClientPool implements ClientPool<IMetaStoreClient, TException
     this.clientPoolCache =
         Caffeine.newBuilder()
             .expireAfterAccess(evictionInterval, TimeUnit.MILLISECONDS)
-            .removalListener((ignored, value, cause) -> ((HiveClientPool) value).close())
+            .removalListener(
+                (ignored, value, cause) -> {
+                  // We have closed the HiveClientPool that is manually removed from the cache.
```

Comment:
The comment is really hard to understand...

Reply:
Modified and clarified it.